runtime/interpreter/value.go

```diff
@@ -2870,6 +2884,37 @@ func (v *ArrayValue) Slice(
 	)
 }
 
+func (v *ArrayValue) Reverse(
+	interpreter *Interpreter,
+	locationRange LocationRange,
+) Value {
+	count := v.Count()
+	idx := count - 1
+
+	return NewArrayValueWithIterator(
+		interpreter,
+		NewVariableSizedStaticType(interpreter, v.Type.ElementType()),
```

Comment:
The type here can also be `ConstantSizedStaticType`. Instead of constructing a new type, can simply pass `v.Type`.